app/controllers/api/v1/communications_controller.rb

```diff
@@ -10,7 +10,7 @@ def show; end
       private
 
       def set_communication
-        @communication = Communication.published.find(params[:id])
```

Comment:
Just create a `not_recurrent` scope and chain them instead of creating an overly specific scope. So it would be `Communication.published.non_recurrent.find(params[:id])`